### ISSUE TITLE
Taint analysis attributes a named argument to its positional index, not the named parameter

### ISSUE THREAD

**Issue by alies-dev** [vimeo/psalm#11923]
### Summary

In taint analysis, a **named argument** is attributed to the parameter at its **positional index** rather than to the parameter it actually names. This produces a false positive on the parameter that happens to sit at that index, and a matching false negative on the parameter that really received the value.

Type checking is unaffected: it uses the correctly resolved `$function_param`. Only the taint graph mis-routes.

### Reproduction

Psalm 7.0.0-beta19, PHP 8.4, no plugins, `errorLevel="1"`, `--taint-analysis`.

```php
<?php declare(strict_types=1);

namespace Min;

/** @psalm-taint-source input */
function tainted(): string { return 'attacker'; }

function sinkNamed(string $path = 'safe', string $label = 'x'): void
{
    echo $label;                   // html sink  -- SHOULD report
    echo file_get_contents($path); // file sink  -- should NOT report
}

function sinkPositional(string $path = 'safe', string $label = 'x'): void
{
    echo $label;                   // html sink  -- SHOULD report
    echo file_get_contents($path); // file sink  -- should NOT report
}

function callNamed(): void      { sinkNamed(label: tainted()); }
function callPositional(): void { sinkPositional('safe', tainted()); }
```

The two sinks are byte-identical and receive the same tainted value; they differ only in whether the caller writes the argument positionally or by name. (They must be separate functions, since taint on a parameter aggregates across all callers.)

### Actual

```
d3_min.php:11:28 - TaintedSSRF: Detected tainted network request
d3_min.php:11:28 - TaintedFile: Detected tainted file handling
d3_min.php:16:10 - TaintedHtml: Detected tainted HTML
d3_min.php:16:10 - TaintedTextWithQuotes: Detected tainted text with possible quotes
```

- `sinkNamed`: reports on **`$path`** (line 11) — false positive — and reports **nothing** on `$label` (line 10) — false negative.
- `sinkPositional`: reports on `$label` (line 16) — correct.

### Expected

Both functions report `TaintedHtml` on `$label` and neither reports `TaintedFile` / `TaintedSSRF` on `$path`.

### Analysis

In `ArgumentsAnalyzer::checkArgumentsMatch()`, the named argument is resolved to the right parameter, but stored under the caller's positional index:

https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentsAnalyzer.php#L793-L816

```php
$arg_function_params[$argument_offset] = [$candidate_param];
```

`$argument_offset` is then forwarded to `ArgumentAnalyzer::checkArgumentMatches()` alongside the correctly matched `$function_param`:

https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentsAnalyzer.php#L865-L875

and the taint node is built from that positional offset:

```php
$method_node = DataFlowNode::getForMethodArgument(
    $cased_method_id,
    $cased_method_id,
    $argument_offset,          // positional index, not the named parameter's index
    $taint_flow_graph ? $function_param->location : null,
    $function_call_location,
);
```

https://github.com/vimeo/psalm/blob/master/src/Psalm/Internal/Analyzer/Statements/Expression/Call/ArgumentAnalyzer.php#L1774-L1791

Note the node already mixes the two: the id uses the positional `$argument_offset` while the location comes from the named `$function_param`. The sink side keys `taint_sink_params` by the declared parameter index, so the two only agree when a named argument happens to be written at its own position.

A fix would pass the matched parameter's declared index (e.g. `array_search`/the index of `$candidate_param` within `$function_params`) into the node construction, leaving `$argument_offset` for diagnostics that genuinely mean "the Nth written argument".

### Impact

Named arguments are idiomatic in modern PHP, and any framework that re-spreads them (`f(...$args)`) propagates the mis-attribution downstream. Reported against `psalm-plugin-laravel` as a `TaintedFile` false positive where the only request-derived value was passed as a named argument to a *different* parameter: https://github.com/psalm/psalm-plugin-laravel/issues/1395

### Versions

- vimeo/psalm 7.0.0-beta19 (`7e751c06a756fa64dc4c759c09fe4a173afcb433`)
- PHP 8.4

**Comment by alies-dev** [comment]
Also affects the 6.x line — reproduced on **Psalm 6.16.1** (`f1f5de594dc76faf8784e02d3dc4716c91c6f6ac`), no plugins, `errorLevel="1"`, `--taint-analysis --no-cache --threads=1`, with the same two-function repro:

```
FILE: src/min.php
| ERROR | 7  | TaintedFile           | Detected tainted file handling             |
| ERROR | 10 | TaintedHtml           | Detected tainted HTML                      |
| ERROR | 10 | TaintedTextWithQuotes | Detected tainted text with possible quotes |
```

Line 7 is `sinkNamed`'s `file_get_contents($path)` — the false positive on the positional parameter; line 6 (`echo $label`, the named target) stays silent — the false negative. Lines 10/11 are `sinkPositional`, correct. `TaintedSSRF` is absent only because `file_get_contents` gained the `ssrf` sink in 7.x; the parameter-level verdict is identical.

The source shape is the same in 6.16.1, at slightly different lines:

- `ArgumentsAnalyzer.php:796-817` — `$arg_function_params[$argument_offset] = [$candidate_param];`
- `ArgumentsAnalyzer.php:868-874` — forwards `$argument_offset + $i` into `checkArgumentMatches()`
- `ArgumentAnalyzer.php:1792-1799` — `DataFlowNode::getForMethodArgument(..., $argument_offset, ...)`

The only difference is cosmetic (`$data_flow_graph instanceof TaintFlowGraph` where 7.x uses `getTaintFlowGraphWithSuppressed()`), so a fix here would likely need a 6.x port as well.
